proto/v1/cel-service.proto

```diff
@@ -0,0 +1,132 @@
+// a server for this API.  The API will be used for conformance testing,
+// utilities, and execution as a service.
+service CelService {
+  // Transforms CEL source text into a parsed representation.
+  rpc Parse (ParseRequest) returns (ParseResponse);
+  // Runs static checks on a parsed CEL representation and return
+  // an annotated representation, or a set of issues.
+  rpc Check (CheckRequest) returns (CheckResponse);
+  // Evaluates a parsed or annotation CEL representation given
+  // values of external bindings.
+  rpc Eval (EvalRequest) returns (EvalResponse);
+}
+
+// Warnings or errors in service execution are represented by
+// [google.rpc.Status] [] messages, with the following message
+// in the details field.
+message StatusDetails {
+  // Severities of issues.
+  enum Severity {
+    // An unspecified severity.
+    SEVERITY_UNSPECIFIED = 0;
+    // Deprecation issue for statements and method that may no longer be
+    // supported or maintained.
+    DEPRECATION = 1;
+    // Warnings such as: unused variables.
+    WARNING = 2;
+    // Errors such as: unmatched curly braces or variable redefinition.
+    ERROR = 3;
+  }
+
+  // The severity of the issue.
+  Severity severity = 1;
+
+  // The following fields specify location, if known.
+
+  // The 1-based index of the starting line in the source text
+  // where the issue occurs, or 0 if unknown.
+  int32 line = 2;
+  // The 0-based index of the starting position within the line of source text
+  // where the issue occurs.  Only meaningful if line is nonzer.
+  int32 column = 3;
+  // Expression ID from [Expr] [], 0 if unknown.
+  int64 id = 4;
+}
+
+// Request message for the Parse method.
+message ParseRequest {
+  // Required. Source text in CEL syntax.
+  string cel_source = 1;
+  // Tag for version of CEL syntax, for future use.
+  string syntax_version = 2;
+  // File or resource for source text, used in [SourceInfo] [].
+  string source_location = 3;
+  // Prevent macro expansion.  See "Macros" in Language Defiinition.
+  bool disable_macros = 4;
+}
+
+// Response message for the Parse method.
+message ParseResponse {
+  // The parsed representation, or unset if parsing failed.
+  ParsedExpr parsed_expr = 1;
+  // Any number of issues with [StatusDetails] [] as the details.
+  repeated google.rpc.Status issues = 2;
+}
+
+// Request message for the Check method.
+message CheckRequest {
+  // Required. The parsed representation of the CEL program.
+  ParsedExpr parsed_expr = 1;
+  // Declarations of types for external variables and functions.
+  // Required if program uses external variables or functions
+  // not in the default environment.
+  repeated google.api.expr.checked.v1.Decl type_env = 2;
+  // The protocol buffer context.  See "Name Resolution" in the
+  // Language Definition.
+  string container = 3;
+  // If false, add standard definitions to the type environment.
+  // See "Standard Definitions" in the Language Definition.
```

Comment:
Maybe write this in the affirmative: 

```
If true, configure an empty environment. If false, configure the environment with
the standard functions and types listed in the spec (default).
```

Reply:
It's a little awkward either way, since writing for the affirmative conflicts with writing for the default, but I've adapted your wording:
  // If true, use only the declarations in [type_env] [].  If false (default), add
  // declarations for the standard definitions to the type environment.  See
  // "Standard Definitions" in the Language Definition.

Reply:
That works for me.